cytoscape-matterjs.js

```diff
@@ -0,0 +1,428 @@
+          cy.fit( options.padding );
+        }
+
+        if( firstUpdate ){
+          // indicate the initial positions have been set
+          layout.trigger('layoutready');
+          firstUpdate = false;
+        }
+      };
+
+      // update the node positions when notified from the thread but
+      // rate limit it a bit (don't want to overwhelm the main/ui thread)
+      thread.on('message', function( e ){
+        var nodeJsons = e.message;
+        nodeJsons.forEach(function( n ){ id2pos[n.data.id] = n.position; });
+
+        if( !updateTimeout ){
+          updateTimeout = setTimeout( function(){
+            update();
+            updateTimeout = null;
+          }, options.refreshInterval );
+        }
+      });
+
+      // we want to keep the json sent to threads slim and fast
+      var eleAsJson = function( ele ){
+        return {
+          data: {
+            id: ele.data('id'),
+            source: ele.data('source'),
+            target: ele.data('target'),
+            parent: ele.data('parent')
+          },
+          group: ele.group(),
+          position: ele.position()
+
+          // maybe add calculated data for the layout, like edge length or node mass
+        };
+      };
+
+      // data to pass to thread
+      var pass = {
+        eles: eles.map( eleAsJson ),
+        refreshIterations: options.refreshIterations
+        // maybe some more options that matter to the calculations here ...
+      };
+
+      // then we calculate for a while to get the final positions
+      thread.pass( pass ).run(function( pass ){
+        var getPos = _ref_('getPos');
+        var broadcast = _ref_('broadcast');
+        var nodeJsons = pass.eles.filter(function(e){ return e.group === 'nodes'; });
+
+        console.log(pass.eles);
+        // calculate for a while (you might use the edges here)
+        for( var i = 0; i < 100000; i++ ){
+          nodeJsons.forEach(function( nodeJson, j ){
+            nodeJson.position = getPos( j, nodeJson );
+          });
+
+          if( i % pass.refreshIterations === 0 ){ // cheaper to not broadcast all the time
+            broadcast( nodeJsons ); // send new positions outside the thread
+          }
+        }
+      }).then(function(){
+        // to indicate we've finished
+        layout.trigger('layoutstop');
+      });
+
+      return this; // chaining
+    };
+
+    Layout.prototype.stop = function(){
+      // continuous/asynchronous layout may want to set a flag etc to let
+      // run() know to stop
+
+      if( this.thread ){
+        this.thread.stop();
+      }
+      console.log('asdf');
+      this.trigger('layoutstop');
+
+      return this; // chaining
+    };
+
+    Layout.prototype.destroy = function(){
+      // clean up here if you create threads etc
+
+      if( this.thread ){
+        this.thread.stop();
+      }
+
+      return this; // chaining
+    };
+
+    cytoscape( 'layout', 'matterjs', Layout ); // register with cytoscape.js
+
+  };
+
```

Comment:
For the cases that follow, you need to pull in the `matter-js` package for CJS, detect global for global case etc.

See Cola for an example: https://github.com/cytoscape/cytoscape.js-cola/blob/master/cytoscape-cola.js#L539